File: c++/2073.cpp
// Link: https://leetcode.com/problems/time-needed-to-buy-tickets/description/
// Runtime: 0 ms
// Memory: 9.31 MB

class Solution 
{
public:
    int timeRequiredToBuy(const std::vector<int>& tickets, int k) 
    {
        int totalTime = 0;
        for (int i = 0; i < static_cast<int>(tickets.size()); i++)
            if (i <= k)
                totalTime += std::min(tickets[i], tickets[k]);
            else
                totalTime += std::min(tickets[i], tickets[k] - 1);
        
        return totalTime;
    }
};

File: python/2073.py
# Link: https://leetcode.com/problems/time-needed-to-buy-tickets/description/
# Runtime: 38 ms
# Memory: 16.56 MB


class Solution:
    def timeRequiredToBuy(self, tickets: List[int], k: int) -> int:
        total_time = 0
        for i, ticket in enumerate(tickets):
            if i <= k:
                total_time += min(ticket, tickets[k])
            else:
                total_time += min(ticket, tickets[k] - 1)

        return total_time
